api/test/model_device.py

```diff
@@ -92,3 +96,12 @@ def test_DeviceRemovedCorrectly(self):
     def test_DevicesCannotHaveSameName(self):
         with self.assertRaisesRegex(Exception, "There is already a device with this name."):
             self.devices.add_device(self.house1id, None, "Kitchen Thermostat", "thermostat", 1, None, "example")
+
+    def test_EnergenieDeviceAddedCorrectly(self):
+        device = self.devices.get_device_by_id(self.socket_id)
+        self.assertEqual(device['configuration']['device_id'], '46865', 'incorrect inserted device id')
+
+    def test_EnergenieDeviceReadState(self):
+        socket = self.devices.get_device_by_id(self.socket_id)
+        current_state = socket.read_current_state()
+        self.assertEquals(current_state['data']['data']['device_type'], 'control', 'state not read correctly')
```

Comment:
Oops, that is what the mihome4u api device_type is for a monitor/control radio device. I'll change this to 'socket'. The values for all of their devices can be seen here: https://mihome4u.co.uk/docs/api-documentation/subdevices-api/register-a-new-subdevice.
I specified this as this test pulls the data for the socket then verifies that the returned data contains correct info. Alternatively I guess I could just check that the error is none.